crates/consensus-logic/src/csm/worker.rs

```diff
@@ -376,12 +376,22 @@ fn apply_action(
 
         // Update checkpoint entry in database to mark it as included in L1.
         SyncAction::UpdateCheckpointInclusion {
-            epoch,
+            checkpoint,
             l1_reference,
         } => {
-            let Some(mut ckpt_entry) = ckpt_db.get_checkpoint_blocking(epoch)? else {
-                warn!(%epoch, "missing checkpoint we wanted to set the state of, ignoring");
-                return Ok(());
+            let epoch = checkpoint.batch_info().epoch();
+
+            let mut ckpt_entry = match ckpt_db.get_checkpoint_blocking(epoch)? {
+                Some(c) => c,
+                None => {
+                    warn!(%epoch, "missing checkpoint we wanted to set the state of, creating new entry");
```

Comment:
I don't really think this is a warning to be honest.

Also, what should we do in the case of reorgs?  This won't be an issue here but maybe we should create a ticket if there isn't a simple and obviously correct way to deal with it.

Reply:
Yes, this isn't a warning anymore. Corrected.

In case of reorgs, we should probably clear the corresponding database checkpoint entries(and possibly other entries which might be related). Just reverting L1 manifests and corresponding tx entries won't be enough.